### PR TITLE
mlnx-ofa_kernel and mlx-tools: inital import

### PR DESCRIPTION
**Pull request by rlmenge**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

Core, HW and ULPs kernel modules

OFED modules. SRPM can be found at https://linux.mellanox.com/public/repo/bluefield/3.9.0/extras/mlnx_ofed/5.6-1.0.3.3/

###### Change Log  

- Import mlnx-ofa_kernel modules

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Pipeline build id: local build

### CODE REVIEW

File: SPECS/mlnx-tools/mlnx-tools.spec [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+Summary:        Mellanox userland tools and scripts
+Name:           mlnx-tools
+Version:        5.2.0
+Release:        2%{?dist}
+License:        CPL 1.0 or BSD or GPLv2
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+Group:          System Environment/Programming
+URL:            https://github.com/Mellanox/mlnx-tools
+Source0:        https://linux.mellanox.com/public/repo/bluefield/3.9.0/extras/mlnx_ofed/5.6-1.0.3.3/SOURCES/mlnx-tools_5.2.0.orig.tar.gz#/%{name}-%{version}.tar.gz
```
Comment (PawelWMS) [inline SPECS/mlnx-tools/mlnx-tools.spec:38]:
```suggestion
Source0:        https://linux.mellanox.com/public/repo/bluefield/3.9.0/extras/mlnx_ofed/5.6-1.0.3.3/SOURCES/%{name}_%{version}.orig.tar.gz#/%{name}-%{version}.tar.gz
```

File: SPECS/mlnx-tools/mlnx-tools.spec [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+Obsoletes: mlnx-ofa_kernel < 5.4, mlnx_en-utils < 5.4
```
Comment (PawelWMS) [inline SPECS/mlnx-tools/mlnx-tools.spec:39]: One per line, please. I think the linter should take care of this.

File: SPECS/mlnx-tools/mlnx-tools.spec [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+%description
+Mellanox userland tools and scripts
+
+%define debug_package %{nil}
+%global PYTHON3 1
```
Comment (PawelWMS) [inline SPECS/mlnx-tools/mlnx-tools.spec:44]: Is this even used?
```suggestion
```

File: SPECS/mlnx-ofa_kernel/mlnx-ofa_kernel.spec [hunk 4]
```diff
@@ -0,0 +1,562 @@
...
+%global POWERKVM %(if (grep -qiE "powerkvm" /etc/issue /etc/*release* 2>/dev/null); then echo -n '1'; else echo -n '0'; fi)
+%global BLUENIX %(if (grep -qiE "Bluenix" /etc/issue /etc/*release* 2>/dev/null); then echo -n '1'; else echo -n '0'; fi)
+%global XENSERVER65 %(if (grep -qiE "XenServer.*6\.5" /etc/issue /etc/*release* 2>/dev/null); then echo -n '1'; else echo -n '0'; fi)
+
+# MarinerOS 1.0 sets -fPIE in the hardening cflags
+# (in the gcc specs file).
+# This seems to break only this package and not other kernel packages.
+%if "%{_vendor}" == "mariner"
+%global _hardened_cflags %{nil}
+%endif
```
Comment (PawelWMS) [inline SPECS/mlnx-ofa_kernel/mlnx-ofa_kernel.spec:49]: I'd drop the `%if` and check, if this even still applies in 2.0.

**Review by hbeberman — APPROVED** [review]

**Pull request merged**